app/assets/javascripts/discourse/app/components/light-dark-img.gjs

```diff
@@ -0,0 +1,54 @@
+import Component from "@glimmer/component";
+import { inject as service } from "@ember/service";
+import CdnImg from "discourse/components/cdn-img";
+import { getURLWithCDN } from "discourse-common/lib/get-url";
+
+export default class LightDarkImg extends Component {
+  @service session;
+
+  get isDarkImageAvailable() {
+    return (
+      this.args.lightImg?.url && // the light image must be present
+      this.args.darkImg?.url &&
+      (this.session.defaultColorSchemeIsDark || this.session.darkModeAvailable)
+    );
+  }
+
+  get defaultImg() {
+    // use dark logo by default in edge case
+    // when scheme is dark and dark logo is present
+    if (this.session.defaultColorSchemeIsDark && this.args.darkImg) {
+      return this.args.darkImg;
+    }
+
+    return this.args.lightImg;
+  }
+
+  get darkImgCdnSrc() {
+    return getURLWithCDN(this.args.darkImg.url);
+  }
+
+  <template>
+    {{#if this.isDarkImageAvailable}}
+      <picture>
+        <source
```

Comment:
Any reasons we can't use the `<CdnImg` component here instead of relying on `darkImgCndSrc`? 🤔

Reply:
Because we're using it in a `<source>` tag. `<CdnImg>` is a `<img>`